resolver.go

```diff
@@ -136,7 +143,13 @@ func (r *resolver) Stop() {
 }
 
 func (r *resolver) SetExtServers(dns []string) {
-	r.extDNS = dns
+	l := len(dns)
+	if l > maxExtDNS {
```

Comment:
We could rewrite the following like this. Avoids unnecessary bounds checking and if conditions:

```
for i, dnsE := range dns[:maxExtDNS] {
    r.extDNSList[i].ipStr = dnsE
}
```